test/test_torch.py

```diff
@@ -6397,80 +6398,210 @@ def test_logical_and(self, device):
     def test_logical_or(self, device):
         self._test_logical(device, 'logical_or', [10, 0, 1, 0], [1, 0, 0, 10], [1, 0, 1, 1])
 
-    def test_clamp(self, device):
+    def generate_clamp_baseline(self, device, min_vals=None, max_vals=None, with_nans=False):
```

Comment:
Would it be easier to just compare directly with NumPy? There's a function _np_compare, in TestTorch that might be useful. Comparing with NumPy also ensures we're compatible with it (although our type promotion rules are different because we bias towards float types; that is, in Numpy float16 x int64 is float64, but in PyTorch it's float16).

Reply:
Yes, I think it would slightly more compact. I kept it this way at first to make sure the tests are the same as they were before. I can refactor them further now. Thanks for the note on type promotion, that's useful to know :)